plone/app/caching/purge.py

```diff
@@ -192,7 +192,7 @@ def fieldFilter():
             assignable = IBehaviorAssignable(self.context, None)
             for behavior in assignable.enumerateBehaviors():
                 if behavior.marker:
-                    new_fields = getFieldsInOrder(behavior.marker)
+                    new_fields = getFieldsInOrder(behavior.interface)
```

Comment:
Shouldn't also the line above be changed (`if behavior.marker` -> `if behavior.interface`)?
Anyway it seems to me that you are totally right about using `behavior.interface` instead of `behavior.marker`.

Reply:
I did not check the code, but if @ale-rt is right and there is a bug in the code above, then the test should not pass.

Reply:
Well marker defaults to interface when it is not set:
https://github.com/plone/plone.behavior/blob/a74bf50b69084407715a4753229b255e03e0d156/plone/behavior/metaconfigure.py#L110
No surprise tests are passing

Reply:
So glad you both took a look! 
For behaviors without marker I committed https://github.com/plone/plone.app.caching/pull/55/commits/3ffeaf177c4de712c432d8b0a6f0de106dad9b76

Cause as I think to understand, marker is the way to apply behavior to a single object without applying it to all instances of a dexterity contenttype. So behavior.marker should not be set and is not set to behavior.interface if not provided. 

Example: default behavior IShortName
behavior.marker	 None
behavior.interface	 <SchemaClass plone.app.dexterity.behaviors.id.IShortName>
getFieldsInOrder(behavior.interface): ('id', <zope.schema._field.ASCIILine object at 0x10c5cf0b8>)